pkg/build/builder/timing/context.go

```diff
@@ -14,23 +17,72 @@ var timingKey key
 
 // NewContext returns a context initialised for use
 func NewContext(ctx context.Context) context.Context {
-	return context.WithValue(ctx, timingKey, &[]buildapi.StageInfo{})
+	return context.WithValue(ctx, timingKey, &[]buildapiv1.StageInfo{})
 }
 
 // fromContext returns the existing data stored in the context
-func fromContext(ctx context.Context) *[]buildapi.StageInfo {
-	return ctx.Value(timingKey).(*[]buildapi.StageInfo)
+func fromContext(ctx context.Context) *[]buildapiv1.StageInfo {
+	return ctx.Value(timingKey).(*[]buildapiv1.StageInfo)
 }
 
 // RecordNewStep adds a new timing step to the context
-func RecordNewStep(ctx context.Context, stageName buildapi.StageName, stepName buildapi.StepName, startTime metav1.Time, endTime metav1.Time) {
+func RecordNewStep(ctx context.Context, stageName buildapiv1.StageName, stepName buildapiv1.StepName, startTime metav1.Time, endTime metav1.Time) {
 	stages := fromContext(ctx)
-	newStages := buildapi.RecordStageAndStepInfo(*stages, stageName, stepName, startTime, endTime)
+	newStages := RecordStageAndStepInfo(*stages, stageName, stepName, startTime, endTime)
 	*stages = newStages
 }
 
 // GetStages returns all stages and steps currently stored in the context
-func GetStages(ctx context.Context) []buildapi.StageInfo {
+func GetStages(ctx context.Context) []buildapiv1.StageInfo {
 	stages := fromContext(ctx)
 	return *stages
 }
+
+// RecordStageAndStepInfo records details about each build stage and step
+func RecordStageAndStepInfo(stages []buildapiv1.StageInfo, stageName buildapiv1.StageName, stepName buildapiv1.StepName, startTime metav1.Time, endTime metav1.Time) []buildapiv1.StageInfo {
```

Comment:
Straight move?

Reply:
other than changing it to use external api objects, yeah.